tests/main/test_core.py

```diff
@@ -41,7 +41,9 @@ def __init__(self, sender: str, subject: str, santa_email: str, message: str):
         self.santa_email = santa_email
         self.message = message
 
-    def __eq__(self, other: str):
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, str):
+            return NotImplemented
```

Comment:
you need to look at the whole change, it was flagged by `mypy`, but it was a shortcut before regardless

Reply:
I see, reverting it and running `mypy` gives a good explanation on why it is unsafe to override a method with a more specific argument type. I guess they suggest `NotImplemented` because that is less invasive to the behavior of the program and `__eq__` might actually figure out an alternative way that works to compare two objects.